### ISSUE TITLE
March 2019 - Scheduled content update

### ISSUE THREAD

**Issue by DawnmarieDesJardins** [Microsoft/MCW-Internet-of-Things#13]
@ZoinerTejada @joelhulen @kylebunting
Solliance team -
This workshop is scheduled for an update in March. Please update this issue with suggested content changes. Once done, we'll assign to our SME team for review and additional feedback.

**Comment by joelhulen** [comment]
Here are my suggestions for an update to this MCW:

- Use Device Stream Workflows to protect communication with IoT devices by placing them behind a firewall
- Cloud-to-device communications to control temperature or turn device off/on
- Implement Logic App to send a notification if a device has been offline for too long (builds on current device outage monitoring in Power BI)

**Comment by DawnmarieDesJardins** [comment]
Previous suggestion from Lead via email:
Rename and refresh to include Remote Monitoring and AAD.

**Comment by joelhulen** [comment]
I like the idea of Remote Monitoring and AAD. However, that's already prominently a part of the IoT for Business MCW. Moreover, IoT for Business will most likely be updated to directly use the Remote Monitoring starter to do the initial provisioning. The bits it's currently using are preview versions and were directly referenced rather than going through the new automatic provisioning process. That said, I don't think it makes sense to have both labs use that fast start solution. I like the fact that this one has a more manual process of adding and monitoring devices. I think that the changes I proposed will help round out the current scenario.

**Comment by darrylhogan** [comment]
I'm happy with the proposed updates.

**Comment by DawnmarieDesJardins** [comment]
@joelhulen @ZoinerTejada 
Solliance team - Please move forward with the update with your recommendations and feedback given.  Thanks!

**Comment by DawnmarieDesJardins** [comment]
@joelhulen 
Hi Joel, 
As you're working through this update, please see the two new open issues #14 and #15 
Thanks Dawnmarie

**Comment by DawnmarieDesJardins** [comment]
Updated Development Timeline
Suggested updates due: 2/27/19 (complete)
SME review: 2/28/19 – 3/1/19 (complete)
Draft due: 3/29/19
SME review: 4/1/19 – 4/10/19
Final due: 4/12/19
Publish: 4/16/19

**Comment by joelhulen** [comment]
@darrylhogan @olivierbloch @iizotov @DawnmarieDesJardins I am ready to wrap this up today, but I've hit a roadblock. In the WDS as well as in the hands-on lab ([Exercise 5](https://github.com/Microsoft/MCW-Internet-of-Things/blob/master/Hands-on%20lab/HOL%20step-by-step%20-%20Internet%20of%20Things.md#exercise-5-reporting-device-outages-with-iot-hub-operations-monitoring)) there is a process of detecting which devices have been disconnected for a certain period of time, and reporting those devices. Up until now, IoT Hub Operations Monitoring has been used for this. As you probably know, Operations Monitoring was removed on March 10th. I've not found a suitable replacement for this for monitoring device connections and disconnections in **real-time** while **maintaining the order of events**. One option I experimented with was to turn on diagnostics and have those piped to Event Hubs, after which point we filter using Stream Analytics. The problem I've found with this approach is that there is still a lag, and those events sometimes appear out of order, invalidating it as a replacement for our original approach. The next path I took was to use Event Triggering, which sends Event Grid events on device connect and disconnect. The downside to this is that it restricts the region attendees can use to only a handful. This also complicates the solution by adding temporary storage of these events (such as Table storage or Cosmos DB) as well as an intermediate processor such as Azure Functions.

The guidance I am seeking is whether to go the Event Grid route, or another I haven't considered, or to remove this part of the lab/WDS until a suitable replacement to Operations Monitoring becomes available? Another option is to drop this from the lab (given the current region limitation) and just have it as part of the WDS.
